Zend/zend_ini.c

```diff
@@ -585,8 +625,18 @@ static zend_ulong zend_ini_parse_quantity_internal(zend_string *value, zend_ini_
 		/* ZEND_STRTOUL() does not report a range error when the subject starts
 		 * with a minus sign, so we check this here. Ignore "-1" as it is
 		 * commonly used as max value, for instance in memory_limit=-1. */
-		if (digits[0] == '-' && !(digits_end - digits == 2 && digits_end == str_end && digits[1] == '1')) {
-			overflow = true;
+		if (is_negative) {
+			if (digits_end - digits == 1 && digits_end == str_end && digits[0] == '1') {
+				retval = -1;
+			} else {
+				overflow = true;
+			}
+		}
+	}
+
+	if (signed_result == ZEND_INI_PARSE_QUANTITY_SIGNED) {
+		if (is_negative) {
+			retval = (zend_ulong) (-1 * (zend_long) retval);
```

Comment:
We could use the `-` operator for this, as I think this is what strtol() uses (as implied by the spec), so it would reduce the risk of a behavior change here:

```
retval = 0u-retval;
```